src/Symfony/Bundle/Resources/config/metadata/resource.xml

```diff
@@ -78,5 +70,10 @@
         <service id="api_platform.cache.metadata.resource_collection" parent="cache.system" public="false">
             <tag name="cache.pool" />
         </service>
+
+        <!-- GraphQl specific metadata factory used in the FieldsBuilder and decorated by doctrine metadata factories -->
+        <service id="api_platform.graphql_metadata.resource.metadata_collection_factory" class="ApiPlatform\GraphQl\Metadata\Factory\GraphQlNestedOperationResourceMetadataFactory">
```

Comment:
**thought:** Shouldn't it be in `graphql.xml`?

Reply:
we can't as the other services need this to be available to decorate it even if graphql is not enabled, or I need to add graphql specific services for doctrine and mongodb too.